pkg/core/resources/apis/mesh/mesh_validator.go

```diff
@@ -165,8 +167,10 @@ func validateDatadog(cfgStr *structpb.Struct) validators.ValidationError {
 		verr.AddViolation("address", "cannot be empty")
 	}
 
-	if cfg.Port > 0xFFFF || cfg.Port < 1 {
-		verr.AddViolation("port", "must be in the range 1 to 65535")
+	if elms := strings.Split(cfg.Address, ":"); len(elms) < 2 {
```

Comment:
Does this need to support IPv6 literals?

Reply:
I am not entirely sure. If it does, how do you suggest we format / validate this?

Reply:
IIUC the IPv6 literal with port would be `[::1]:80`, so probably sufficient to split on the last ":" in the string, then you may have to manually remove the square brackets.

Reply:
I think [SplitHostPort](https://pkg.go.dev/net#SplitHostPort) does what you want here. I just happened across it :)

Reply:
We're using `govalidator` package in the project, it has a lot of useful methods, one of them is `govalidator.IsURL`, maybe it's better to use it here

Reply:
Nice find with SplitHostPort! I think that's exactly what we need here.

Reply:
Also, looks like we already use that in `validateLoggingTcp` in the same file for the same purpose.